docker-proper.go

```diff
@@ -140,22 +147,24 @@ func getExpired(client *dockerclient.DockerClient) (expiredContainers []*dockerc
 			continue
 		}
 		debug("  + older than %s", *ageImage)
-		if usedImages[image.Id] > 0 {
-			debug("  + in use, skipping")
-			continue
-		}
 		expiredImages = append(expiredImages, image)
 	}
+
 	log.Printf("Found %d expired images", len(expiredImages))
-	return expiredContainers, expiredImages, nil
-}
 
-func removeImages(client *dockerclient.DockerClient, images []*dockerclient.Image) error {
-	for _, image := range images {
+	for _, image := range expiredImages {
 		log.Print("rmi ", image.Id)
 		if !*dry {
 			if _, err := client.RemoveImage(image.Id, false); err != nil {
-				debug("Couldn't remove image: %s", err)
+				debug("Couldn't remove image with ID %s. Let's try to remove the same image, but this time using repo tags: %s", image.Id, err)
+			}
+		}
+		for _, repoTag := range image.RepoTags {
```

Comment:
if it successfully removed, do you still need this loop? 
will you try to remove something that's gone? and i guess that'll fail too, which is fine...